### DISCUSSION TITLE
How to Get Mouse Coordinates on Stage and Its Speed for Extensions??

### DISCUSSION THREAD

**Discussion by hammouda101010** [TurboWarp/extensions#1714] [Q&A] [answered]
i dont think there is an extension for this. but i need a way to fetch the mouse cords from the stage and convert it into speed. is there a way to fetch something like this?

**Comment by David-Orangemoon** [comment, marked as answer]
```js
vm.renderer.canvas.addEventListener("mousemove",()=>{/*eventCode*/});
vm.renderer.canvas.addEventListener("mouseover",()=>{/*eventCode*/});
```
https://developer.mozilla.org/en-US/docs/Web/API/Element/mousemove_event
https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseover_event
Here are the docs for those events.

**Reply by hammouda101010** [reply]
ok can you show an example
